opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -296,277 +196,151 @@ class Restart : private boost::noncopyable
-    }
-
-    // no destructor in this class as ERT takes care of freeing the
-    // handle as part of freeing the solution handle!
-
-    ecl_sum_tstep_type *ertHandle() const
-    { return ertHandle_; };
-
-private:
-    ecl_sum_tstep_type *ertHandle_;
-};
-
-
 /**
  * Initialization file which contains static properties (such as
  * porosity and permeability) for the simulation field.
  */
-class Init : private boost::noncopyable
-{
+class Init {
 public:
-    Init(const std::string& outputDir,
-         const std::string& baseName,
-         int writeStepIdx,
-         IOConfigConstPtr ioConfig) : egridFileName_(outputDir,
-                                                     baseName,
-                                                     ECL_EGRID_FILE,
-                                                     writeStepIdx,
-                                                     ioConfig->getFMTOUT())
-    {
-        bool formatted = ioConfig->getFMTOUT();
-
-        FileName initFileName(outputDir,
-                              baseName,
-                              ECL_INIT_FILE,
-                              writeStepIdx,
-                              formatted);
-
-        ertHandle_ = fortio_open_writer(initFileName.ertHandle(),
-                                        formatted,
-                                        ECL_ENDIAN_FLIP);
-    }
+    Init( const std::string& outputDir,
+          const std::string& baseName,
+          int writeStepIdx,
+          const IOConfig& ioConfig ) :
+        egrid( outputDir,
+               baseName,
+               ECL_EGRID_FILE,
+               writeStepIdx,
+               ioConfig.getFMTOUT() ),
+        init( fortio_open_writer(
+                    FileName( outputDir,
+                        baseName,
+                        ECL_INIT_FILE,
+                        writeStepIdx,
+                        ioConfig.getFMTOUT() ).ertHandle(),
+                    ioConfig.getFMTOUT(),
+                    ECL_ENDIAN_FLIP ) )
 
-    ~Init()
-    { fortio_fclose(ertHandle_); }
+    {}
 
-    void writeHeader(int numCells,
-                     const int* compressedToCartesianCellIdx,
-                     const SimulatorTimerInterface& timer,
-                     Opm::EclipseStateConstPtr eclipseState,
-                     const PhaseUsage uses,
-                     const NNC& nnc = NNC())
+    void writeHeader( int numCells,
+                      const int* compressedToCartesianCellIdx,
+                      time_t current_posix_time,
+                      const EclipseState& es,
+                      int ert_phase_mask,
+                      const NNC& nnc = NNC() )
     {
-        auto dataField = eclipseState->get3DProperties().getDoubleGridProperty("PORO").getData();
-        restrictAndReorderToActiveCells(dataField, numCells, compressedToCartesianCellIdx);
+        auto dataField = es.get3DProperties().getDoubleGridProperty("PORO").getData();
```

Comment:
note that outputing *anything* directly from EclipseState's grid properties should be considered to be bad style because these properties may be inconsistent with what the simulator actually uses: in this case (PORO), the simulator actually uses the pore volume which may or may not be calculated using PORO; in any case, the porosity gets modified by the rock compressibility, NTG and MULTPV, so the values which are put into the output files might not be what the user expects...